File: dao/src/main/java/org/thingsboard/server/dao/settings/MailTemplateService.java
// SPDX-FileCopyrightText: Copyright The Thingsboard Authors
// SPDX-License-Identifier: Apache-2.0
package org.thingsboard.server.dao.settings;

import org.thingsboard.server.common.data.id.TenantId;
import org.thingsboard.server.common.data.mail.MailTemplateSettings;

public interface MailTemplateService {

    MailTemplateSettings getMailTemplateSettings(TenantId tenantId);

    MailTemplateSettings saveMailTemplateSettings(TenantId tenantId, MailTemplateSettings settings);

}

File: ui-ngx/src/app/core/http/mail-template.service.ts
// SPDX-FileCopyrightText: Copyright The Thingsboard Authors
// SPDX-License-Identifier: Apache-2.0
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

import { defaultHttpOptionsFromConfig, RequestConfig } from '@core/http/http-utils';
import { MailTemplateSettings } from '@shared/models/mail-template.models';

@Injectable({
  providedIn: 'root'
})
export class MailTemplateService {

  constructor(private http: HttpClient) {}

  public getMailTemplateSettings(config?: RequestConfig): Observable<MailTemplateSettings> {
    return this.http.get<MailTemplateSettings>('/api/tenant/mailTemplates', defaultHttpOptionsFromConfig(config));
  }

  public saveMailTemplateSettings(settings: MailTemplateSettings, config?: RequestConfig): Observable<MailTemplateSettings> {
    return this.http.post<MailTemplateSettings>('/api/tenant/mailTemplates', settings, defaultHttpOptionsFromConfig(config));
  }

}

